src/java.base/share/classes/sun/util/locale/BaseLocale.java

```diff
@@ -164,11 +164,17 @@ public static BaseLocale getInstance(String language, String script,
         // "interned" instance can subsequently be used by the Locale
         // instance which guarantees the locale components are properly cased/interned.
         return CACHE.intern(new BaseLocale(language, script, region, variant),
-            (b) -> new BaseLocale(
-                LocaleUtils.toLowerString(b.language).intern(),
-                LocaleUtils.toTitleString(b.script).intern(),
-                LocaleUtils.toUpperString(b.region).intern(),
-                b.variant.intern()));
+                // Avoid lambdas since this may be on the bootstrap path in many locales
+                new UnaryOperator<BaseLocale>() {
```

Comment:
Since this intern is called many times, can we convert this `new` to a local class in a static method and reuse its singleton?